tests/FSharp.Compiler.ComponentTests/Conformance/BasicGrammarElements/CustomAttributes/CallerArgumentExpression/CallerArgumentExpression.fs

```diff
@@ -0,0 +1,384 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Conformance.BasicGrammarElements
+
+open Xunit
+open FSharp.Test.Compiler
+open FSharp.Test
+
+module CustomAttributes_CallerArgumentExpression =
+
+    [<FactForNETCOREAPP>]
+    let ``Can consume CallerArgumentExpression in BCL methods`` () =
+      let path = __SOURCE_DIRECTORY__ ++ "test script.fsx"
+      FsFromPath path
+      |> withLangVersionPreview
+      |> asExe
+      |> compileAndRun
+      |> shouldSucceed
+      |> ignore
+
+    [<FactForNETCOREAPP>]
+    let ``Can define methods using CallerArgumentExpression with C#-style optional arguments`` () =
+        FSharp """let assertEqual a b = if a <> b then failwithf "not equal: %A and %A" a b
+open System.Runtime.CompilerServices
+open System.Runtime.InteropServices
+type A() =
+  static member aa (
+    a,
+    [<CallerMemberName; Optional; DefaultParameterValue "no value">]b: string, 
+    [<CallerLineNumber; Optional; DefaultParameterValue 0>]c: int, 
+    [<CallerArgumentExpressionAttribute("a"); Optional; DefaultParameterValue "no value">]e: string) = 
+    a,b,c,e
+
+let stringABC = "abc"
+assertEqual (A.aa(stringABC)) ("abc", ".cctor", 13, "stringABC")
+assertEqual (A.aa(a = stringABC)) ("abc", ".cctor", 14, "stringABC")
+        """
+        |> withLangVersionPreview
+        |> asExe
+        |> compileAndRun
+        |> shouldSucceed
+        |> ignore
+
+    [<FactForNETCOREAPP>]
+    let ``Can define methods using CallerArgumentExpression with F#-style optional arguments`` () =
+        FSharp """let assertEqual a b = if a <> b then failwithf "not equal: %A and %A" a b
+open System.Runtime.CompilerServices
+open System.Runtime.InteropServices
+type A() =
+  static member aa (
+    a,
+    [<CallerMemberName>] ?b: string, 
+    [<CallerLineNumber>] ?c: int, 
+    [<CallerArgumentExpressionAttribute("a")>] ?e: string) = 
```

Comment:
Does F# style optional argument also support combination of `CallerArgumentExpressionAttribute` and `Struct` to generate a `ValueOption` member in IL ?